src/ctia/lib/riemann.clj

```diff
@@ -0,0 +1,124 @@
+                                   :status (str (:status response))}))
+          response)
+        (catch ExceptionInfo e
+          (let [data (ex-data e)
+                ex-type (:type data)
+                evt {:metric (ms-elapsed start)
+                     :service (str service-name " error")
+                     :description (.getMessage e)
+                     :error "true"
+                     :status (condp = ex-type
+                               :ring.util.http-response/response (get-in data [:response :status])
+                               :compojure.api.exception/response-validation "500"
+                               :compojure.api.exception/request-validation "400"
+                               "500")}]
+            (send-request-metrics send-event-fn request evt))
+          (throw e))
+        (catch Throwable e
+          (send-request-metrics send-event-fn request
+                                {:metric (ms-elapsed start)
+                                 :service (str service-name " error")
+                                 :description (.getMessage e)
+                                 :stacktrace (.getStackTrace e)
+                                 :status "500"
+                                 :error "true"})
+          (throw e))))))
+
+(defn prepare-event [event]
+  {:user-id (get-in event [:user :id])})
+
+(defn send-event
+  [conn prefix event]
+  (let [prepared-event (-> event
+                           prepare-event
+                           (update :service #(str prefix " " %)))]
+    (if conn
+      (do
+        (log/debugf "Sending event to riemann:\n%s\nprepared:\n%s"
+                    (utils/safe-pprint event)
+                    (utils/safe-pprint prepared-event))
+        (riemann/send-event conn prepared-event))
+      (when-not (= "log" (:event-type prepared-event))
+        (log/warnf "Riemann doesn't seem configured. Event: %s"
+                   (utils/safe-pprint prepared-event))))))
+
+(defn wrap-request-logs
+  "Middleware to log all incoming connections to Riemann"
+  [service-name]
+  (let [{enabled? :enabled :as config}
+        (get-in @prop/properties [:ctia :log :riemann])]
+    (if-not enabled?
+      identity
+      (let [_ (log/info "riemann metrics reporting")
+            send-event-fn 
+            (let [client (riemann/tcp-client
+                           (select-keys config
+                                        [:host :port :interval-in-ms]))]
+              (fn [event]
+                (riemann/send-event client event)))]
+        (fn [handler]
+          (wrap-request-metrics handler
+                                service-name
+                                send-event-fn))))))
+
+(defn log
+  "Produce a log and send an event to riemann.
+  The event should contains the fields :level to specify the log level.
+  The r-service parameter should be the service name that appear in riemann."
+  [_ r-service msg event]
+  '...)
```

Comment:
I don't think you need to create log function for riemann in the PR.
More than that. We intend to use riemann directly via `timbre` and use the `timbre/with-context` to really improve our logging API inside IROH. And This is a bit refactorisation on its own for CTIA too.